commit 8c4c74ae15c3eaff9e1d25ba749d161181284570
Author: Yifeng Li <tomli@tomli.me>
Date:   Thu May 4 01:34:14 2017 +0800

    rt286: add Thinkpad Helix 2 to force_combo_jack_table
    
    commit fe0dfd6358a17c79bd7d6996af7512ba452a7059 upstream.
    
    Thinkpad Helix 2 is a tablet PC, the audio is powered by Core M
    broadwell-audio and rt286 codec. For all versions of Linux kernel,
    the stereo output doesn't work properly when earphones are plugged
    in, the sound was coming out from both channels even if the audio
    contains only the left or right channel. Furthermore, if a music
    recorded in stereo is played, the two channels cancle out each other
    out, as a result, no voice but only distorted background music can be
    heard, like a sound card with builtin a Karaoke sount effect.
    
    Apparently this tablet uses a combo jack with polarity incorrectly
    set by rt286 driver. This patch adds DMI information of Thinkpad Helix 2
    to force_combo_jack_table[] and the issue is resolved. The microphone
    input doesn't work regardless to the presence of this patch and still
    needs help from other developers to investigate.
    
    This is my first patch to LKML directly, sorry for CC-ing too many
    people here.
    
    Link: https://bugzilla.kernel.org/show_bug.cgi?id=93841
    Signed-off-by: Yifeng Li <tomli@tomli.me>
    Signed-off-by: Mark Brown <broonie@kernel.org>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

diff --git a/sound/soc/codecs/rt286.c b/sound/soc/codecs/rt286.c
index 9c365a7..7899a2c 100644
--- a/sound/soc/codecs/rt286.c
+++ b/sound/soc/codecs/rt286.c
@@ -1108,6 +1108,13 @@ static const struct dmi_system_id force_combo_jack_table[] = {
 			DMI_MATCH(DMI_PRODUCT_NAME, "Kabylake Client platform")
 		}
 	},
+	{
+		.ident = "Thinkpad Helix 2nd",
+		.matches = {
+			DMI_MATCH(DMI_SYS_VENDOR, "LENOVO"),
+			DMI_MATCH(DMI_PRODUCT_VERSION, "ThinkPad Helix 2nd")
+		}
+	},
 
 	{ }
 };
